src/TTImp/ProcessBuiltin.idr

```diff
@@ -129,45 +185,82 @@ checkCons c cons ty fc = case !(foldr checkCon (pure (Nothing, Nothing)) cons) o
         , builtinTransforms.natSNames $= insert cons.succ MkSUCC
         }
 
+addNatToInteger :
+    Ref Ctxt Defs =>
+    (fn : Name) ->
+    NatToInt ->
+    Core ()
+addNatToInteger fn nToI = do
+    log "builtin.NaturalToInteger.addTransforms" 10
+        $ "Add %builtin NaturalToInteger transform for " ++ show fn ++ "."
+    update Ctxt $ record
+        { builtinTransforms.natToIntegerFns $= insert fn nToI
+        }
+
 ||| Check a `%builtin Natural` pragma is correct.
 processBuiltinNatural :
-    {auto c : Ref Ctxt Defs} ->
-    {auto m : Ref MD Metadata} ->
-    {auto u : Ref UST UState} ->
+    Ref Ctxt Defs =>
     Defs -> FC -> Name -> Core ()
 processBuiltinNatural ds fc name = do
-    log "builtin.Natural" 5 $ "Processing Builtin Natural pragma for " ++ show name
+    log "builtin.Natural" 5 $ "Processing %builtin Natural " ++ show name ++ "."
     [(n, _, gdef)] <- lookupCtxtName name ds.gamma
         | [] => throw $ UndefinedName fc name
         | ns => throw $ AmbiguousName fc $ (\(n, _, _) => n) <$> ns
     let TCon _ _ _ _ _ _ dcons _ = gdef.definition
-        | def => throw $ GenericMsg fc $ "Expected a type constructor, found:\n" ++ show def
+        | def => throw $ GenericMsg fc
+            $ "Expected a type constructor, found " ++ showDefType def ++ "."
     cons <- getConsGDef ds.gamma fc dcons
-    cons <- checkCons ds.gamma cons n fc
+    cons <- checkNatCons ds.gamma cons n fc
     zero <- getFullName cons.zero
     succ <- getFullName cons.succ
-    n <- getFullName name
     addBuiltinNat n $ MkNatBuiltin {zero, succ}
 
+||| Check a `%builtin NaturalToInteger` pragma is correct.
+processNatToInteger :
+    Ref Ctxt Defs =>
+    Defs -> FC -> Name -> Core ()
+processNatToInteger ds fc fn = do
+    log "builtin.NaturalToInteger" 5 $ "Processing %builtin NaturalToInteger " ++ show fn ++ "."
+    [(n, _, gdef)] <- lookupCtxtName fn ds.gamma
+        | [] => throw $ UndefinedName fc fn
+        | ns => throw $ AmbiguousName fc $ (\(n, _, _) => n) <$> ns
+    let PMDef _ args _ cases _ = gdef.definition
+        | def => throw $ GenericMsg fc
+            $ "Expected function definition, found " ++ showDefType def ++ "."
+    let [(_ ** arg)] = getNEArgs gdef.type
+        | [] => throw $ GenericMsg fc $ "No arguments found for " ++ show n ++ "."
+        | _ => throw $ GenericMsg fc $ "More than 1 non-erased arguments found for " ++ show n ++ "."
+    let Just tyCon = getTypeCons arg
+        | Nothing => throw $ GenericMsg fc
+            $ "No type constructor found for non-erased arguement of " ++ show n ++ "."
+    Just _ <- getNatBuiltin tyCon
+        | Nothing => throw $ GenericMsg fc $ "Non-erased argument is not a 'Nat'-like type."
+    let arity = length $ getTypeArgs gdef.type
+    let Just natIdx = getNEIndex arity gdef.type
+        | Nothing => throw $ InternalError "Couldn't find non-erased argument."
+    addNatToInteger n (MkNatToInt {arity, natIdx})
+    pure ()
```

Comment:
No need to finish a block with `pure ()`

Reply:
Leftover from when I was defining it to get it to typecheck, and I forgot to remove it...